src/analyzer/analyzersilence.cpp

```diff
@@ -10,14 +10,36 @@ namespace {
 // verify that the track samples have not changed since the last analysis
 constexpr CSAMPLE kSilenceThreshold = 0.001f; // -60 dB
 // TODO: Change the above line to:
-//constexpr CSAMPLE kSilenceThreshold = db2ratio(-60.0f);
+// constexpr CSAMPLE kSilenceThreshold = db2ratio(-60.0f);
+
+// These are in dBV expressed as Volts RMS (which seems, sensibly,
+// the way Mixxx works).
+// Don't change these to const as they are used only to feed the
+// fade thresholds which are consts below themselves while we work
+// out which one is best, and you'll just be adding to exe size.
+#define N_10DB_FADEOUT_THRESHOLD 0.3162f
+#define N_12DB_FADEOUT_THRESHOLD 0.2511f
+#define N_15DB_FADEOUT_THRESHOLD 0.1778f
+#define N_18DB_FADEOUT_THRESHOLD 0.1259f
+#define N_20DB_FADEOUT_THRESHOLD 0.1f
+#define N_24DB_FADEOUT_THRESHOLD 0.0631f
+#define N_25DB_FADEOUT_THRESHOLD 0.0562f
+#define N_27DB_FADEOUT_THRESHOLD 0.0447f
+#define N_30DB_FADEOUT_THRESHOLD 0.0316f
+#define N_40DB_FADEOUT_THRESHOLD 0.01f
```

Comment:
make them `constexpr`. they have internal linkage and the compiler will strip them out.

Reply:
No. I want the compiler to strip them out.  Only a couple of these are used, and then only as part of a const.  They are there just while we(?) work out the best values. This makes it easier to change. They can be deleted later when we have a fixed value.

Reply:
> No. I want the compiler to strip them out. Only a couple of these are used, and then only as part of a const. 

Right, [as long as you mark them `const`. the compiler will remove them.](https://compiler-explorer.com/#g:!((g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:1,endLineNumber:9,positionColumn:1,positionLineNumber:9,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:'namespace+%7B%0Aconstexpr+int+internal1+%3D+1%3B%0A%7D%0Aconstexpr+int+internal2+%3D+2%3B%0Aconst+int+internal3+%3D+3%3B%0A%0A//+only+this+shows+up+in+the+binary%0Aint+external+%3D+3%3B%0A'),l:'5',n:'1',o:'C%2B%2B+source+%231',t:'0')),k:50,l:'4',m:50,n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(compilerName:'x86-64+gcc+14.2',editorid:1,fontScale:14,fontUsePx:'0',j:1,wrap:'1'),l:'5',n:'0',o:'Output+of+x86-64+gcc+(trunk)+(Compiler+%231)',t:'0')),header:(),l:'4',m:50,n:'0',o:'',s:0,t:'0')),k:50,l:'3',n:'0',o:'',t:'0'),(g:!((h:compiler,i:(compiler:gsnapshot,filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1',verboseDemangling:'0'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!(),options:'',overrides:!(),selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x86-64+gcc+(trunk)+(Editor+%231)',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4).

Anyways, not particularly important if these are only meant to be temporary (that was not clear from the source).

Reply:
>>  while we work out which one is best